### PR TITLE
cloud-init: add impish integration tests

### PR DESCRIPTION
**Pull request by orndorffgrant**
I went ahead and did a first pass at this, just following the patterns of past releases. I'm not sure how to test it though. Let me know if I missed anything.

### CODE REVIEW

File: jenkins-jobs/cloud-init/integration-ec2.yaml [hunk 1]
```diff
@@ -95,6 +95,22 @@
       - integration-ec2:
           release: hirsute
 
+- job:
+    name: cloud-init-integration-ec2-i
+    defaults: cloud-init
+    node: torkoal
+    triggers:
+      - timed: "H 0 * * 6"
```
Comment (paride) [inline jenkins-jobs/cloud-init/integration-ec2.yaml:103]: We run the `ec2` jobs on different days, so we don't exceed the limits of our ec2 account and to be sure that `cloud-init-integration-ec2-terminate` does not terminate instances that are still being used by another project.

Let's make the trigger:
```
timed: "H 0 * * 3"
```
as day 3 is not used by other ec2 jobs.

(The better way to do this is with the "throttle concurrent builds" plugin, which is installed but we need IS to create a job category for these ec2 jobs. At the moment I'm doing it for curtin, if it works fine I'll also ask for a cloud-init-ec2 category.)

**Review by paride — CHANGES_REQUESTED** [review]
Thanks! There is just one little thing to change, see the inline comment.

File: jenkins-jobs/cloud-init/integration-ec2.yaml (hunk 1 above)
Comment (orndorffgrant) [reply jenkins-jobs/cloud-init/integration-ec2.yaml:103]: Ah I didn't notice that, and thanks for the explanation! I'll make this change quick

Comment (orndorffgrant) [reply jenkins-jobs/cloud-init/integration-ec2.yaml]: @paride Just updated it!

Comment (paride) [reply jenkins-jobs/cloud-init/integration-ec2.yaml]: Thanks! Will merge after CI passes.

**Review by paride — APPROVED** [review]
LGTM!

**Pull request merged**